### ISSUE TITLE
aQute.bnd.osgi.Instruction must prevent error when a empty input string is passed

### ISSUE THREAD

**Issue by cvgaviao** [bndtools/bnd#2544]
While trying to test the bndlib 4.1.0-SNAPSHOT with maven-bundle-plugin I found an error in one of its unit test.
The error is occurring because m-b-p is mounting this string:

     *;classifier=;type=jar;scope=compile,*;classifier=;type=jar;scope=runtime

 and calling Instruction("") constructor with a empty input for the classifier item. 
```
java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at java.lang.String.charAt(String.java:658)
	at aQute.bnd.osgi.Instruction.<init>(Instruction.java:86)
	at org.apache.felix.bundleplugin.AbstractDependencyFilter$DependencyFilter.<init>(AbstractDependencyFilter.java:86)
	at org.apache.felix.bundleplugin.AbstractDependencyFilter$DependencyFilter.<init>(AbstractDependencyFilter.java:80)
	at org.apache.felix.bundleplugin.AbstractDependencyFilter$7.<init>(AbstractDependencyFilter.java:243)
	at org.apache.felix.bundleplugin.AbstractDependencyFilter.processInstructions(AbstractDependencyFilter.java:243)
	at org.apache.felix.bundleplugin.DependencyEmbedder.processHeaders(DependencyEmbedder.java:92)
	at org.apache.felix.bundleplugin.BundlePlugin.addMavenInstructions(BundlePlugin.java:715)
	at org.apache.felix.bundleplugin.BundlePlugin.buildOSGiBundle(BundlePlugin.java:969)
	at org.apache.felix.bundleplugin.BundlePluginTest.testEmbedDependencyPositiveClauses(BundlePluginTest.java:240)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at junit.framework.TestCase.runTest(TestCase.java:176)
	at junit.framework.TestCase.runBare(TestCase.java:141)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:129)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:538)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:760)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:460)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:206)

```

**Comment by bjhargrave** [comment]
That should never have worked. Instruction has always required a non-empty string input. So something else is wrong.

The test code https://github.com/apache/felix/blob/bb983d28e43be0b55fb86af8b70477439f1ff3be/tools/maven-bundle-plugin/src/test/java/org/apache/felix/bundleplugin/BundlePluginTest.java#L235-L236 

        instructions.put( DependencyEmbedder.EMBED_DEPENDENCY, "*;classifier=;type=jar;scope=compile,"
            + "*;classifier=;type=jar;scope=runtime" );

shows `*` in there. Some one on the maven-bundle-plugin team will need to debug this further.

**Comment by bjhargrave** [comment]
Looking at the code some more, it seems to be because `classifier=`. That is, `classifier` is set to the empty string and this empty string is used to create an Instruction. But as I said, `new Instruction("")` have never worked.

**Comment by cvgaviao** [comment]
@bjhargrave, strange, some how it have been working with bndlib 3.5.0.  

So, I tried to prevent m-b-p to pass an empty string to Instruction() but doing that causes another problem: Embedded-Artifacts and Bundle-ClassPath are including bundles using classifier different than empty. 

I will try another approach...

**Comment by bjhargrave** [comment]
Fixed in 4.1.
